activesupport/lib/active_support/core_ext/time/calculations.rb

```diff
@@ -166,6 +166,10 @@ def change(options)
     else
       ::Time.new(new_year, new_month, new_day, new_hour, new_min, new_sec, utc_offset)
     end
+
+    # if hour isn't changing on a daylight savings day, keep the original hour
+    second_repeated_dst_hour = options[:hour].nil? && (self - 1.hour).hour == hour
+    second_repeated_dst_hour ? new_time + 1.hour : new_time
```

Comment:
This check is expensive. Since UTC times are very common and not concerned, it would be best to avoid this computation in such case.

I'd also not use ternaries here, as it really doesn't help readability.